tron/bin/get_tron_metrics.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python3.6
+#
+# get_tron_metrics.py
+#   This script is designed to retrieve metrics from Tron via its API and send
+#   send them to meteorite.
+import logging
+import pprint
+import subprocess
+import sys
+import textwrap
+
+from tron.commands import cmd_utils
+from tron.commands.client import Client
+
+log = logging.getLogger('get_tron_metrics')
+
+
+def parse_cli():
+    parser = cmd_utils.build_option_parser()
+    parser.description = (
+        "Collects metrics from Tron via its API and forwards them to "
+        "meteorite."
+    )
+    parser.add_argument(
+        "--dry-run",
+        action="store_true",
+        default=False,
+        help="Don't actually send metrics out. Defaults: %(default)s"
+    )
+    args = parser.parse_args()
+    return args
+
+
+def check_bin_exists(bin):
+    """
+    Checks if an executable binary exists
+
+    :param bin: (str) Name of the executable; could be a path to one
+    """
+    try:
+        subprocess.call(['which', bin])
+    except OSError:
+        return False
+    return True
+
+
+def send_data_metric(
+    name, metric_type, value,
+    dimensions={}, dry_run=False, **kwargs
+):
+    """
+    Sends a single data point to meteorite via bash command
+
+    :param name: (str) Name of the metric
+    :param metric_type: (str) Type of the meteorite metric. Must be in
+        METEORITE_TYPES
+    :param value: (float) Value of the metric
+    :param dimensions: (dict) Metric dimensions as key-value pairs
+    :param dry_run: (bool) Whether or not to send metrics to meteorite
+    """
+    if dry_run:
+        metric_args = dict(
+            name=name,
+            metric_type=metric_type,
+            value=value,
+            dimensions=dimensions,
+        )
+        log.info(
+            f"Would have sent this to meteorite:\n"
+            f"{pprint.pformat(metric_args)}"
+        )
+        return
+
+    cmd = ['meteorite', 'data', '-v', name, metric_type, str(value)]
+    for k, v in dimensions.items():
+        cmd.extend(['-d', f"{k}:{v}"])
+
+    process = subprocess.Popen(
+        cmd,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.PIPE,
+    )
+    output, error = process.communicate()
+    output = output.decode('utf-8').rstrip()
+    error = error.decode('utf-8').rstrip()
+
+    if process.returncode:
+        log.error(
+            "Meteorite failed with:\n"
+            f"{textwrap.indent(error, '    ')}"
+        )
+    else:
+        log.debug(f"From meteorite: {output}")
+
+
+def send_counter(name, **kwargs):
+    send_data_metric(name, 'counter', kwargs.pop('count'), **kwargs)
```

Comment:
When you're actually calling send_data_metric, can you specify the kwargs, instead of passing everything you get with **kwargs? Also use the keyword names, like metric_type='counter'.

Reply:
Will change.